test/fuzz/fuzz_util.h

```diff
@@ -0,0 +1,47 @@
+/*-
+ * Public Domain 2014-2020 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+typedef struct {
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+} FUZZ_GLOBAL_STATE;
+
+typedef struct {
+    const uint8_t **slices;
+    size_t *sizes;
+    size_t num_slices;
+} FUZZ_SLICED_INPUT;
+
+extern FUZZ_GLOBAL_STATE fuzz_state;
+
+void fuzzutil_setup();
+bool fuzzutil_sliced_input_init(FUZZ_SLICED_INPUT *input, const uint8_t *data, size_t size,
```

Comment:
The design looks a bit different from the regular WiredTiger style.

That's because returning non-zero due to bad input is not ok. It's not a failure and we don't want to record it, we just want to return out. Therefore, there isn't much value in having an API that returns error codes because we can't really use `WT_RET` and WT_ERR` the same way.

Also, our fuzz target code has to use non-WiredTiger stuff like raw `malloc` which doesn't conform to our API design so the code already doesn't follow our conventions.